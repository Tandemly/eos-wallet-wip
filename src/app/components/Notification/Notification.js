import * as React from "react";
import cx from "classnames";
import css from "./styles.module.scss";

const Notification = ({ text, status }) => (
  !status ? null :
  <div className={cx(css.notification, css[status])}>
    <div className={cx("icon-success", css.icon)}></div>
    {text}
    <span className={cx("icon-x", css.close)}></span>
  </div>
);

export default Notification;
