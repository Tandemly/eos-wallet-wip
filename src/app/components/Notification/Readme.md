### Notification examples

An error notificaiton

```js

<Background className="notification-example content">
  <Notification
    text="Cannot connect to the Wallet"
    status="error"
  />
</Background>
```

A success notificaiton

```js

<Background className="notification-example content">
  <Notification
    text="Transaction to joesmith successful"
    status="success"
  />
</Background>
```
