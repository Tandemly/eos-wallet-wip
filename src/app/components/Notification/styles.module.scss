@import "../../styles/custom_variables.scss";

.notification {
  width: 100%;
  border-radius: 8px;
  color: #fff;
  margin-bottom: 1.5rem;
  padding: 0.875rem 3rem 0.875rem 4.5rem;
  position: relative;

  .icon {
    width: 50px;
    height: 100%;
    position: absolute;
    left: 0;
    top: 0;
    border-radius: 8px 0 0 8px;
    text-align: center;
    line-height: 50px;
    font-size: 1.5rem
  }

  .close {
    position: absolute;
    top: 0.875rem;
    right: 1rem;
  }
}

.error {
  background-color: #574750;
  border: 1px solid $red;

  .icon {
    background-color: $red;
  }
}

.success {
  background-color: #475753;
  border: 1px solid $green;

  .icon {
    background-color: $green;
    color: $grey-80;
  }
}