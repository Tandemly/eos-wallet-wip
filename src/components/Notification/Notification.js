import * as React from "react";
import cx from "classnames";
import css from "./styles.module.scss";

const Notification = ({ text, status, unsetNotification }) => (
  !status ? null :
  <div className={cx(css.notification, css[status])}>
    <div className={cx("icon-success", css.icon)}></div>
    {text}
    <span
      className={cx("icon-x", css.close)} 
      onClick={unsetNotification}
    />
  </div>
);

export default Notification;
