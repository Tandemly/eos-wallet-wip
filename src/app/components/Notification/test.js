import * as React from "react";
import renderer from 'react-test-renderer';
import Notification from './';

describe('<Notification />', () => {
  it('renders without crashing', () => {
    const notification = {
      text: 'Connection error',
      status: 'error'
    };
    const tree = renderer.create(
      <Notification {...notification} />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
})


