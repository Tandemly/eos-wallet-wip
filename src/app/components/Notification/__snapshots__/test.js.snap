// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Notification /> renders without crashing 1`] = `
<div
  className=""
>
  <div
    className="icon-success"
  />
  Connection error
  <span
    className="icon-x"
  />
</div>
`;
